invokeai/app/services/shared/sqlite_migrator/migrations/migration_4.py

```diff
@@ -0,0 +1,94 @@
+import sqlite3
+
+from invokeai.app.services.shared.sqlite_migrator.sqlite_migrator_common import Migration
+
+
+class Migration4Callback:
+    """Callback to do step 4 of migration."""
+
+    def __call__(self, cursor: sqlite3.Cursor) -> None:  # noqa D102
+        self._create_model_metadata(cursor)
+        self._create_model_tags(cursor)
+        self._create_tags(cursor)
+        self._create_triggers(cursor)
+
+    def _create_model_metadata(self, cursor: sqlite3.Cursor) -> None:
+        """Create the table used to store model metadata downloaded from remote sources."""
+        cursor.execute(
+            """--sql
+            CREATE TABLE IF NOT EXISTS model_metadata (
+                id TEXT NOT NULL,
+                name TEXT GENERATED ALWAYS AS (json_extract(metadata, '$.name')) VIRTUAL NOT NULL,
+                author TEXT GENERATED ALWAYS AS (json_extract(metadata, '$.author')) VIRTUAL NOT NULL,
+                -- Serialized JSON representation of the whole metadata object,
+                -- which will contain additional fields from subclasses
+                metadata TEXT NOT NULL,
+                created_at DATETIME NOT NULL DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')),
+                -- Updated via trigger
+                updated_at DATETIME NOT NULL DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')),
+                FOREIGN KEY(id) REFERENCES model_config(id)
```

Comment:
Right now there is no uniqueness constraint on the `id` field. Is there intended to be a 1-to-1 relationship between `model_config` and `model_metadata`. If so, maybe `id` should be a primary key.

If the relationship is in fact 1-to-1, we may also want to consider merging `model_config` and `model_metadata` into a single table.

Reply:
I've made `id` into a primary key.

As it is, the relationship is 1:Optional. That is, a model can have a config record but no metadata record. If the two tables get merged then there would have to be a lot of NULLable fields, which is probably not desirable.

Reply:
Agreed. If 1:0..1, then this is the right approach.